language/stdlib/modules/libra_account.mvir

```diff
@@ -97,18 +83,17 @@ module LibraAccount {
         // Load the sender's account
         sender = get_txn_sender();
         sender_account_ref = borrow_global_mut<T>(copy(sender));
-        // Log a send event
-        sent_event = SentPaymentEvent { amount: copy(deposit_value), payee: copy(payee) };
+        // Log a Payment event to sender
+        event = PaymentEvent { amount: copy(deposit_value), payee: copy(payee), payer: move(sender) };
 
-        Event.emit_event<Self.SentPaymentEvent>(&mut move(sender_account_ref).sent_events, move(sent_event));
+        Event.emit_event<Self.PaymentEvent>(&mut move(sender_account_ref).payment_events, copy(event));
 
         // Load the payee's account
```

Comment:
To address Ben's feedback, perhaps we could:
- Move the payee logic before the payer logic
- After executing the payee logic, check if the sender is equal to the payee and return before executing the payer logic if so.